hnn_core/network.py

```diff
@@ -1015,6 +1003,116 @@ def add_connection(self, src_gids, target_gids, loc, receptor,
 
         self.connectivity.append(deepcopy(conn))
 
+    def pick_connection(self, src_gids=None, target_gids=None,
+                        loc=None, receptor=None):
```

Comment:
I think this would be better off as a function. So that you don't get the impression that calling this method would *modify* `net.connectivity` and include only the connections that satisfy the filter. Pondering over this led me to this stackoverflow post:

https://stackoverflow.com/questions/8108688/in-python-when-should-i-use-a-function-instead-of-a-method

Let me know what you think ...

Reply:
I'm fine with making this a function! I think my main hesitation against this was having to pass the whole network object as an argument, as something seems fishy about passing such a large object to a function with so many irrelevant attributes.

In any case I agree that the idea of a "search function" is better off being separate from the Network, despite being completely tied to the Network object.